src/Samples/Common/ViewModels/FeatureSamples/CustomPrimitiveTypes/SampleId.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using Newtonsoft.Json;
+
+namespace DotVVM.Samples.Common.ViewModels.FeatureSamples.CustomPrimitiveTypes
+{
+    public record SampleId : TypeId<SampleId>
+    {
+        public SampleId(Guid idValue) : base(idValue)
+        {
+        }
```

Comment:
:thinking: 

We need to be able work with the id server-side and client-side in a similar way. Making the ID record solves comparison. We'd also need to override `ToString()` to be the same as `IdValue.ToString()`, otherwise `{{resource: Id}}` will behave differently than `{{value: Id}}` (even the value binding should evaluate to the same value server-side and client-side).